File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    answer = 0

    for i in range(len(schedules)):
        h = schedules[i] // 100
        m = schedules[i] % 100
        goaltime = (h * 60) + m + 10
        timelog = timelogs[i]
        
        cnt = 0
        day = startday
        for j in range(7):
            now = day % 7
            
            if now == 6 or now == 0:
                day += 1
                continue
            else:
                h = timelog[j] // 100
                m = timelog[j] % 100
                checktime = (h * 60) + m 
                
                if checktime <= goaltime:
                    cnt += 1
                else:
                    break
                    
            if cnt == 5:
                answer += 1 
            day += 1
                           
    return answer

File: 프로그래머스/1/388351. 유연근무제/유연근무제.js
function solution(schedules, timelogs, startday) {
    let answer = 0;
    
    for (let i = 0; i < schedules.length; i++) {
        // 목표 시간 : 출근 희망 시간 + 10분
        const h = Math.floor(schedules[i] / 100);
        const m = schedules[i] % 100;
        const goalTime = h * 60 + m + 10;
        
        let isPass = true;
        
        for (let j = 0; j < 7; j++) {
            // 현재 요일 계산
            const day = ((startday + j - 1) % 7) + 1;
            
            // 주말은 건너뜀
            if (day >= 6) {
                continue;
            }
            
            const logH = Math.floor(timelogs[i][j] / 100);
            const logM = timelogs[i][j] % 100;
            const checkTime = logH * 60 + logM;
            
            // 탈락 여부 판단
            if (checkTime > goalTime) {
                isPass = false;
                break;
            }
        }
        
        if (isPass) {
            answer++;
        }
    }
    return answer;
}